File: packages/api-client/src/jobs.ts
import type { ApiClient } from './client'

export interface JobCreated {
  job_id: number
}

export interface JobStatus {
  status: 'pending' | 'running' | 'done' | 'failed'
  job_type: 'saju_report' | 'compatibility'
  result_id?: number
  error?: string
}

/** GET /api/jobs/{id} — job 상태 폴링 (소유자). */
export function getJob(api: ApiClient, jobId: number): Promise<JobStatus> {
  return api.get<JobStatus>(`/api/jobs/${jobId}`)
}

File: backend/schemas/jobs.py
"""비동기 job API 스키마."""
from __future__ import annotations

from pydantic import BaseModel


class JobCreatedResponse(BaseModel):
    job_id: int


class JobStatusResponse(BaseModel):
    status: str                 # pending | running | done | failed
    job_type: str               # saju_report | compatibility
    result_id: int | None = None
    error: str | None = None
